### PR TITLE
Skip signing VisualFsharp.sln and FSharp.sln when building with .NET …

### PR DESCRIPTION
**Pull request by ellahathaway**
Closes https://github.com/dotnet/fsharp/issues/18112

May need to be adjusted based on https://github.com/dotnet/fsharp/issues/18112#issuecomment-2524557168

When `VisualFSharp.sln` and `FSharp.sln` are built with .NET Core MSBuild, no artifacts are built that can be signed -> signing fails when an empty `ItemsToSign` list.

This PR includes a change that sets `-sign` to false when building `VisualFSharp.sln` or `FSharp.sln` with .NET Core on windows.

### CODE REVIEW

**Comment by ellahathaway** [comment]
Marking this as ready for review. This change is needed to unblock https://github.com/dotnet/sdk/pull/44855. Depending on the feedback to https://github.com/dotnet/fsharp/issues/18112#issuecomment-2524557168, I'm happy to either revise this PR or open a follow up PR later.

**Comment by ellahathaway** [comment]
@dotnet/fsharp-team-msft - friendly ping. Can I please get eyes on this? It's needed to enable signing in the VMR for the unified build project.

File: eng/Build.ps1 [hunk 1]
```diff
@@ -563,12 +563,19 @@ try {
...
+            # don't produce any artifacts to sign. Skip signing in this case.
+            $sign = $False
+        }
         if ($noVisualStudio) {
             BuildSolution "FSharp.sln" $False
         }
         else {
             BuildSolution "VisualFSharp.sln" $False
         }
+        $sign = $originalSignValue
```
Comment (mmitche) [inline eng/Build.ps1:578]: Does the oriignal value need to be restored?

Comment (ellahathaway) [reply eng/Build.ps1:578]: Yeah. It's for the case that the remaining solutions get built as well, eg `FSharp.Benchmarks.sln` and `Microsoft.FSharp.Compiler.sln`

**Review by mmitche — APPROVED** [review]

**Review by psfinaki — APPROVED** [review]

**Comment by psfinaki** [comment]
@ellahathaway sorry for the delay. We are waiting for @KevinRansom to take a look at this as he's most knowledgeable in the topic.

If this is in a hurry && low risk, we can probably merge it as is and hope things work out. Letting @T-Gro to decide tho.

**Review by KevinRansom — APPROVED** [review]

**Comment by KevinRansom** [comment]
@ellahathaway sorry for the delay I was in England.  But now Jet-lag has my attention.

**Pull request merged**